lightning/src/ln/router.rs

```diff
@@ -939,19 +932,23 @@ impl Router {
 					}
 				}
 
-				for chan_id in $node.channels.iter() {
-					let chan = network.channels.get(chan_id).unwrap();
-					if chan.one_to_two.src_node_id == *$node_id {
-						// ie $node is one, ie next hop in A* is two, via the two_to_one channel
-						if first_hops.is_none() || chan.two_to_one.src_node_id != network.our_node_id {
-							if chan.two_to_one.enabled {
-								add_entry!(chan_id, chan.one_to_two.src_node_id, chan.two_to_one, $fee_to_target_msat);
-							}
-						}
-					} else {
-						if first_hops.is_none() || chan.one_to_two.src_node_id != network.our_node_id {
-							if chan.one_to_two.enabled {
-								add_entry!(chan_id, chan.two_to_one.src_node_id, chan.one_to_two, $fee_to_target_msat);
+				if !$node.features.requires_unknown_bits() {
```

Comment:
Sadly we're not going to route through a node which requires a local policy feature like data_loss_protect while this doesn't concern payment path at all. That's a spec wrongdoing